### PR TITLE
Small changes that shouldn't affect how it works

### PR DESCRIPTION
**Pull request by Artoria2e5**
* A caching function is added to the non-approximate mode, so we don't need to call partial() every time.
* The constants are formatted and useless hardcodes are purged completely.

### CODE REVIEW

**Comment by Artoria2e5** [comment]
Okay, now it does affect it a bit. Basically fractionals don't make sense in the color dict and we should just replace them with fixed interger steps.

File: minecraftmap/__init__.py [hunk 1]
```diff
@@ -199,24 +223,27 @@ def tocoord(self,xy):
...
+        lookupindex refers to constants.estimationlookup and can be None or 1 for no approximation.'''
+        # Exact hits, or?
+        if color in self.allcolorsinversemap:
             return self.allcolorsinversemap[color]
-        except KeyError:
-            if self.uselookupdict and lookupindex in constants.estimationlookupdict:
-                return constants.estimationlookupdict[lookupindex][(color[0]*lookupindex//255,color[1]*lookupindex//255,color[2]*lookupindex//255)]
-            elif not self.uselookupdict and lookupindex in constants.estimationlookup:
-                return constants.estimationlookup[lookupindex][color[0]*lookupindex//255][color[1]*lookupindex//255][color[2]*lookupindex//255]
+        elif color in constants.estimationlookupdict
```
Comment (Artoria2e5) [inline minecraftmap/__init__.py]: oh snap
```suggestion
        elif color in constants.estimationlookupdict:

```

File: minecraftmap/__init__.py [hunk 2]
```diff
@@ -199,24 +223,28 @@ def tocoord(self,xy):
...
+        elif color in constants.estimationlookupdict:
+
+            return constants.estimationlookupdict[color]
+        elif color in Map.colorcache:
+            return self.colorcache[color]
+        elif lookupindex is None or lookupindex != 1:
+            # Use interval-approximated approximate. Hard to roll off the tongue.
+            if not constants.hasInterval(lookupindex, self.uselookupdict):
+                constants.addestimate(lookupindex, self.uselookupdict)
+            if self.uselookupdict：
```
Comment (Artoria2e5) [inline minecraftmap/__init__.py]:
```suggestion
            if self.uselookupdict:

```

**Comment by Artoria2e5** [comment]
Feck, the change is big now...

**Comment by Artoria2e5** [comment]
Huh, PIL has a ImagePalette thing that I can use with dithering and all the wonderful shit. Who said we need to build our own wheel again?

File: README.rst [hunk 3]
```diff
@@ -7,12 +7,13 @@ Simple Python 3 editor for Minecraft maps (the item) in NBT format. Allows editi
 
 Usage::
 
+.. highlight:: python
```
Comment (Artoria2e5) [inline README.rst:10]: Nope.

**Comment by greener-dot-ca** [comment]
I ended up rolling the code I needed right into papyri, so fill your boots with this :)

**Comment by Artoria2e5** [comment]
Oh, so you found this a big mess too? Good to know I wasn't getting high on sleep deprivation...

**Comment by Artoria2e5** [comment]
Yeah, I should just publish my own pypi package at this point. Off to writing a test so I don't commit syntax effups again...

**Pull request closed without merging**